vscode/src/auth/auth.ts

```diff
@@ -79,9 +80,10 @@ export async function showSignInMenu(
             // Auto log user if token for the selected instance was found in secret
             const selectedEndpoint = item.uri
             const token = await secretStorage.getToken(selectedEndpoint)
+            const tokenSource = await secretStorage.getTokenSource(selectedEndpoint)
             let { authStatus } = token
                 ? await authProvider.validateAndStoreCredentials(
-                      { serverEndpoint: selectedEndpoint, accessToken: token },
+                      { serverEndpoint: selectedEndpoint, accessToken: token, tokenSource: tokenSource },
```

Comment:
When the property name and the variable name are the same, you can just name the variable.
```suggestion
                      { serverEndpoint: selectedEndpoint, accessToken: token, tokenSource },
```